### PR TITLE
Modify operator to deploy both legacy and csi driver by default.

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

Note that this PR will cause the operator to completely ignore disableCsi, and always deploy both daemonsets. Follow up PRs will allow one to toggle either one on or off.

```release-note
NONE
```

### CODE REVIEW

**Comment by brybacki** [comment]
I think this change deserves a release note. You have release note "NONE".

**Comment by awels** [comment]
Yeah, well the entire release is going to be about switching to CSI. there are several PRs all related to this. So by itself this is not that important, that is why the NONE is there.

File: pkg/controller/hostpathprovisioner/controller.go [hunk 1]
```diff
@@ -419,8 +424,13 @@ func (r *ReconcileHostPathProvisioner) checkDegraded(logger logr.Logger, cr *hos
...
 		return true, err
 	}
+	daemonSetCsi := &appsv1.DaemonSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: fmt.Sprintf("%s-csi", MultiPurposeHostPathProvisionerName), Namespace: namespace}, daemonSet)
+	if err != nil {
+		return true, err
+	}
 
-	if !checkDaemonSetReady(daemonSet) {
+	if !checkDaemonSetReady(daemonSet) || checkDaemonSetReady(daemonSetCsi) {
```
Comment (brybacki) [inline pkg/controller/hostpathprovisioner/controller.go:433]: `!checkDaemonSetReady(daemonSet) || checkDaemonSetReady(daemonSetCsi) `
non csi false and csi true means degraded? shouldn't be 
`!checkDaemonSetReady(daemonSet) || !checkDaemonSetReady(daemonSetCsi) `

**Comment by brybacki** [comment]
Most of it looks like removing some `if !enabled` conditions, there is one thing I do not understand (commented in controller.go). 
Other than that looks good.

File: pkg/controller/hostpathprovisioner/controller.go (hunk 1 above)
Comment (awels) [reply pkg/controller/hostpathprovisioner/controller.go]: Yes, so it turns out I was looking up the wrong daemonset right above. so it should be correct now.

**Comment by awels** [comment]
Remove the whole disableCsi flag in the CR. In a follow up, I will make the legacy provisioner optional (turn it on by default, then after that turn it off by default, then eventually remove it again)

**Pull request merged**